pkg/ocm/accounts_mgmt.go

```diff
@@ -0,0 +1,75 @@
+package ocm
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-openapi/strfmt"
+	amgmtv1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+	"github.com/openshift/assisted-service/pkg/commonutils"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockgen -package ocm -destination mock_accounts_mgmt.go . OCMAccountsMgmt
+type OCMAccountsMgmt interface {
+	CreateSubscription(ctx context.Context, clusterID strfmt.UUID) (*amgmtv1.Subscription, error)
+	UpdateSubscription(ctx context.Context, subscriptionID, openshiftClusterID strfmt.UUID) (*amgmtv1.Subscription, error)
+}
+
+type accountsMgmt struct {
+	client *Client
+}
+
+func (a accountsMgmt) CreateSubscription(ctx context.Context, clusterID strfmt.UUID) (*amgmtv1.Subscription, error) {
+	defer commonutils.MeasureOperation("OCM-CreateSubscription", a.client.log, a.client.metricsApi)()
+
+	//create the request
+	request, err := amgmtv1.NewSubscriptionRegistration().ClusterUUID(clusterID.String()).PlanID("AssistedInstall").Status("Reserved").Build()
+	if err != nil {
+		a.client.logger.Error(context.Background(), "Failed to create subscription registeration request: %v", err)
```

Comment:
You are getting a context as a parameter why do you create a new context?
Same comment for all the contexts used in this funciton

Reply:
Actually I was just copying from authz pkg [here](https://github.com/ybettan/assisted-service/blob/7fa60b0f3839f2227412bcdf1afb37d4f2fd7405/pkg/ocm/authorization.go#L81) but you are right, I should use the propagated ctx, changing it.

Reply:
Where does it write this log?